libcontainer/factory.go

```diff
@@ -27,11 +28,15 @@ var idRegex = regexp.MustCompile(`^[\w+-\.]+$`)
 
 // New returns a linux based container factory based in the root directory.
 func New(root string) (*Factory, error) {
+	absRoot, err := filepath.Abs(root)
+	if err != nil {
+		return nil, err
+	}
 	if err := os.MkdirAll(root, 0o700); err != nil {
 		return nil, err
 	}
 	return &Factory{
-		Root: root,
+		Root: absRoot,
```

Comment:
Wondering why we made `Root` public; after all, its expected to only be set once (when doing `libcontainer.New()`), then add a `.Root()` function.

Once it's private, we should also validate (if needed) `root` here, so that we can remove redundant checks such as
https://github.com/opencontainers/runc/blob/master/libcontainer/factory_linux.go#L145-L147

and

https://github.com/opencontainers/runc/blob/master/libcontainer/factory_linux.go#L220-L222

And remove redundant code, such as https://github.com/opencontainers/runc/blob/6d35069b5ebeb5b5fd33a7778aab368fea7d5030/list.go#L118-L122

Also, not 100% sure about this utility; perhaps it's still needed though (but it's a bit of an odd one)
https://github.com/opencontainers/runc/blob/c9b8b4f35eae6a57f64b5d9b69bd49d133833c34/utils.go#L99-L101

Reply:
Yes, I was going to clean this up.